flocker/node/agents/ebs.py

```diff
@@ -152,6 +218,62 @@ def compute_instance_id(self):
         """
         return get_instance_metadata()['instance-id'].decode("ascii")
 
+    def _get(self, blockdevice_id):
+        """
+        Lookup BlockDeviceVolume for given blockdevice_id.
+
+        :param unicode blockdevice_id: ID of a blockdevice that needs lookup.
+
+        :returns BlockDeviceVolume for the given input id.
+        """
+        for volume in self.list_volumes():
+            if volume.blockdevice_id == blockdevice_id:
+                return volume
+        raise UnknownVolume(blockdevice_id)
+
+    def _get_ebs_volume(self, blockdevice_id):
+        """
+        Lookup EBS Volume information for a given blockdevice_id.
+
+        :param unicode blockdevice_id: ID of a blockdevice that needs lookup.
+
+        :returns boto.ec2.volume.Volume for the input id. ``None`` if
+            no boto.ec2.volume.Volume was found for the given id.
+        """
+        for volume in self.connection.get_all_volumes():
```

Comment:
A minor optimization here might be `get_all_volumes(volume_ids=[blockdevice_id])`.

:question: (optional to address since performance is not our main concern and there's no benchmark demonstrating this actually helps)